paasta_tools/contrib/service_shard_remove.py

```diff
@@ -0,0 +1,141 @@
+import argparse
+import logging
+
+from paasta_tools.config_utils import AutoConfigUpdater
+from paasta_tools.utils import DEFAULT_SOA_CONFIGS_GIT_URL
+from paasta_tools.utils import format_git_url
+from paasta_tools.utils import load_system_paasta_config
+
+log = logging.getLogger(__name__)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="")
+    parser.add_argument(
+        "--git-remote",
+        help="Master git repo for soaconfigs",
+        default=None,
+        dest="git_remote",
+    )
+    parser.add_argument(
+        "--branch", help="Branch name to push to", required=True, dest="branch",
+    )
+    parser.add_argument(
+        "--local-dir",
+        help="Act on configs in the local directory rather than cloning the git_remote",
+        required=False,
+        default=None,
+        dest="local_dir",
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        help="Logging verbosity",
+        action="store_true",
+        dest="verbose",
+    )
+    parser.add_argument(
+        "--source-id",
+        help="String to attribute the changes in the commit message.",
+        required=False,
+        default=None,
+        dest="source_id",
+    )
+    parser.add_argument(
+        "--service", help="Service to modify", required=True, dest="service",
+    )
+    parser.add_argument(
+        "--shard-name", help="Shard name to remove", required=True, dest="shard_name",
+    )
+    parser.add_argument(
+        "--superregion",
```

Comment:
cluster might be a better name (e.g., i’m pretty sure infrastage and kubestage aren’t superregions)

Reply:
this is more of a nit since y’all are probably never gonna run this in the test cluster tho

Reply:
Gotcha, I'm not super familiar with the naming of the hierarchy and how it relates to clusters, but the data source that we are using deals with superregions which is why I chose that here